citesphere/src/main/java/edu/asu/diging/citesphere/core/service/impl/CitationManager.java

```diff
@@ -506,6 +510,27 @@ public CitationPage getPrevAndNextCitation(IUser user, String groupId, String co
     public void deleteLocalGroupCitations(String groupId) {
         citationStore.deleteCitationByGroupId(groupId);
     }
+    
+    @Override
+    public Citations findAuthorityItems(IAuthorityEntry entry, IUser user) {       
+        Citations citations = citationDao.findCitationsByPersonUri(entry.getUri());
+        if (citations != null) {
+            List<ICitation> allCitations = citations.getCitations();
+
+            List<ICitationGroup> groups = getGroups(user);
+            Set<String> groupIds = groups.stream()
+                    .map(group -> group.getKey().toString()) 
+                    .collect(Collectors.toSet());
+
+            List<ICitation> filteredCitations = allCitations.stream()
+                    .filter(citation -> groupIds.contains(citation.getGroup().toString()))
+                    .collect(Collectors.toList());
```

Comment:
this filtering should happen on the db level (when calling `findCitationsByPersonUri`)